sys/event.c

```diff
@@ -127,11 +127,13 @@ RegisterPendingIrpMain(__in PDEVICE_OBJECT DeviceObject, __in PIRP Irp,
   irpEntry->Flags = Flags;
 
   // Update the irp timeout for the entry
+  ExAcquireResourceExclusiveLite(&vcb->Dcb->Resource, TRUE);
```

Comment:
This and associated release call will fail if vcb is null (see vcb check next line).